Allow RunException to fail tests correctly

milljm opened the pull request:
A couple RunException tests were allowed to pass when they should
have been failing. However, we still want these tests to pass, so
use expect_out, instead of expect_error for the deprecated tests
located in misc/check_error.

Closes #10907

Commit:
Allow RunException to fail tests correctly

A couple RunException tests were allowed to pass when they should
have been failing. However, we still want these tests to pass, so
use expect_out, instead of expect_error for the deprecated tests
located in misc/check_error.

Closes #10907

milljm commented on python/TestHarness/testers/RunException.py:
```diff
@@ -67,7 +67,7 @@ def processResults(self, moose_dir, options, output):
 
         if reason == '':
-            RunApp.testFileOutput(self, moose_dir, options, output)
+            reason = RunApp.testFileOutput(self, moose_dir, options, output)
```
woops :)

moosebuild commented:
Job [Documentation](https://www.moosebuild.org/job/160664/) on 06902e4 wanted to post the following:

View the site [here](http://mooseframework.org/docs/PRs/10909/site)

This comment will be updated on new commits.

Commit:
Exit with a non-zero

RunException tests need to exit with a non-zero.

Closes #10907

brianmoose commented on python/chigger/tests/line_sample/line_sample_error.py:
```diff
@@ -18,3 +18,4 @@
 sample.update()
 x = sample[0].getDistance()
 y = sample[0].getSample('invalid_name') # testing this error
+exit(1)
```
@aeslaughter This test was a `RunException` test that requires a non-zero exit status.
It is either this change or change the test spec into a `RunApp` test with `expect_out` and setting `errors` to a dummy value (since there is an `ERROR` in the output). Thoughts?

aeslaughter replied:
Just looking for the error in the output should be fine.

Commit:
line_sample_error should not exit(1) cleanly

Allow line_sample_error to cause an exception, but use RunApp to capture
expect_out. To do this, we need to suppress RunApp's default behavior of
labeling this exception causing test as a failure, by overriding the
errors parameter with a string which will never occur in the output.

Closes #10907

brianmoose commented:
The failing Mac Tests are not related to this PR
